src/components/datagrid/controls/display_selector.tsx

```diff
@@ -318,6 +361,22 @@ export const useDataGridDisplaySelector = (
             )}
           </EuiI18n>
         )}
+        {showResetButton && (
+          <EuiPopoverFooter>
+            <EuiFlexGroup justifyContent="flexEnd" responsive={false}>
+              <EuiFlexItem>
+                <EuiButtonEmpty
+                  flush="both"
+                  size="xs"
+                  onClick={resetToInitialState}
+                  data-test-subj="resetDisplaySelector"
+                >
+                  {resetButtonLabel}
+                </EuiButtonEmpty>
```

Comment:
:sigh: I forgot that immediate children of flex items get expanded, need to wrap the button in a div so it stays small
```suggestion
                <div>
                  <EuiButtonEmpty
                    flush="both"
                    size="xs"
                    onClick={resetToInitialState}
                    data-test-subj="resetDisplaySelector"
                  >
                    {resetButtonLabel}
                  </EuiButtonEmpty>
                </div>
```

Reply:
Doh, I always forget this too. Thanks for checking this. TBH, I wonder if it would be simpler at this point to just drop the flex group completely and add basic inline styles on EuiPopoverFooter:

```js
<EuiPopoverFooter style={{ display: 'flex', justifyContent: 'flex-end' }} />
```

I'll commit this as-is, just thinking out loud